src/hooks/useConditionsList.tsx

```diff
@@ -0,0 +1,61 @@
+import { useQuery } from '@apollo/react-hooks'
+
+import { buildQueryConditionsList } from 'queries/CTEConditions'
+import { Conditions } from 'types/generatedGQLForCTE'
+import { AdvancedFilter, ConditionType, OracleFilterOptions, StatusOptions } from 'util/types'
+
+/**
+ * Return a array of conditions.
+ */
+export const useConditionsList = (advancedFilter: AdvancedFilter) => {
+  const {
+    ConditionType: ConditionTypeFilter,
+    FromToCreationDate,
+    MinMaxOutcomes,
+    ReporterOracle,
+    Status,
+    TextToSearch,
+  } = advancedFilter
+
+  const query = buildQueryConditionsList(advancedFilter)
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const variables: { [k: string]: any } = {}
+  if (ReporterOracle.type === OracleFilterOptions.Custom)
+    variables['oracleNotIn'] = ReporterOracle.value
+  if (
```

Comment:
There must be a better way to build queries and variables, but for the moment this should be accompanied with a test at least. About the type of variables, we can use `[k in someType]: anotherType` , if both types are generic we have a `Record`, I found that it is better to have only some valid keys in variables.

Reply:
I didn't find any other solution or a queryBuilder or a solution that the library gives me.